runtime/src/test/java/com/flipkart/flux/resource/StateMachineResourceTest.java

```diff
@@ -284,6 +284,145 @@ public void testPostScheduledEvent_withTriggerTimeInMilliSeconds() throws Except
 
     }
 
+    @Test
+    public void testPostEventUpdate_withNoEventData() throws Exception {
+        String stateMachineDefinitionJson = IOUtils.toString(this.getClass().getClassLoader().getResourceAsStream("state_machine_definition.json"));
+        Unirest.post(STATE_MACHINE_RESOURCE_URL).header("Content-Type", "application/json").body(stateMachineDefinitionJson).asString();
+
+        Thread.sleep(100);
+        String eventJson = IOUtils.toString(this.getClass().getClassLoader().getResourceAsStream("no_event_data.json"));
+        final HttpResponse<String> eventPostResponse = Unirest.post(STATE_MACHINE_RESOURCE_URL + SLASH + "magic_number_1" + "/context/eventupdate")
+                .header("Content-Type", "application/json").body(eventJson).asString();
+        assertThat(eventPostResponse.getStatus()).isEqualTo(Response.Status.BAD_REQUEST.getStatusCode());
+    }
+
+    @Test
+    public void testEventUpdate() throws Exception {
+        String stateMachineDefinitionJson = IOUtils.toString(this.getClass().getClassLoader().getResourceAsStream(
+                "state_machine_definition.json"));
+        final HttpResponse<String> smCreationResponse = Unirest.post(STATE_MACHINE_RESOURCE_URL).header(
+                "Content-Type", "application/json").body(stateMachineDefinitionJson).asString();
+        Event event = eventsDAO.findBySMIdAndName(smCreationResponse.getBody(), "event0");
+        assertThat(event.getStatus()).isEqualTo(Event.EventStatus.pending);
+
+        /* Make the task fail, eventually sidelined. */
+        TestWorkflow.shouldFail = true;
+        try {
+            /* Since event0 is in pending state, updateEvent should fail. */
+            String eventJson0 = IOUtils.toString(this.getClass().getClassLoader().getResourceAsStream(
```

Comment:
correct eventData should fail as well as task in pending state. Test with valid eventData.